homeassistant/components/matrix/services.py

```diff
@@ -36,20 +42,61 @@
                 MESSAGE_FORMATS
             ),
             vol.Optional(ATTR_IMAGES): vol.All(cv.ensure_list, [cv.string]),
+            vol.Optional(ATTR_THREAD_ID): cv.string,
         },
         vol.Required(ATTR_TARGET): vol.All(
             cv.ensure_list, [cv.matches_regex(CONF_ROOMS_REGEX)]
         ),
     }
 )
 
+compiledEmojiRegex = re.compile(
+    "[\U0001f600-\U0001f64f"  # Emoticons
+    "\U0001f300-\U0001f5ff"  # Symbols & Pictographs
+    "\U0001f680-\U0001f6ff"  # Transport & Map Symbols
+    "\U0001f1e0-\U0001f1ff"  # Flags (iOS)
+    "\U00002700-\U000027bf"  # Dingbats
+    "\U00002600-\U000026ff"  # Miscellaneous Symbols
+    "\U0001f900-\U0001f9ff"  # Supplemental Symbols and Pictographs
+    "\U0001fa70-\U0001faff"  # Symbols and Pictographs Extended-A
+    "\U00002500-\U00002bef"  # Chinese/Japanese/Korean characters
+    "]+",
+    flags=re.UNICODE,
+)
+
+
+def isEmoji(value: Any) -> str:
+    """Validate that value is an emoji."""
+    if not isinstance(value, str):
+        raise vol.Invalid(f"not a string value: {value}")
+
+    if not compiledEmojiRegex.match(value):
+        raise vol.Invalid(f"value {value} is not a valid emoji")
+
+    return value
+
+
+SERVICE_SCHEMA_REACT = vol.Schema(
+    {
+        vol.Required(ATTR_REACTION): isEmoji,
+        vol.Required(ATTR_ROOM): cv.matches_regex(CONF_ROOMS_REGEX),
+        vol.Required(ATTR_MESSAGE_ID): cv.string,
+    }
+)
```

Comment:
Per the matrix spec for annotations (https://spec.matrix.org/v1.9/client-server-api/#events-20), there's no specific requirement that an annotation (aka reaction) needs to be an emoji at all: there are indeed clients out there (fluffychat, schildichat) that fully support custom text, or combinations of emoji (required to support skin tone emoji such as 👌🏾).

I'd say the easiest way to be in compliance with this is to allow arbitrary unicode.

The docs can note that emoji are the common usecase, but should also note that this is not enforced